library/core/src/cell.rs

```diff
@@ -823,11 +843,21 @@ impl<T: ?Sized> RefCell<T> {
     /// ```
     #[stable(feature = "try_borrow", since = "1.13.0")]
     #[inline]
+    #[cfg_attr(feature = "debug_refcell", track_caller)]
     pub fn try_borrow(&self) -> Result<Ref<'_, T>, BorrowError> {
         match BorrowRef::new(&self.borrow) {
-            // SAFETY: `BorrowRef` ensures that there is only immutable access
-            // to the value while borrowed.
-            Some(b) => Ok(Ref { value: unsafe { &*self.value.get() }, borrow: b }),
+            Some(b) => {
+                #[cfg(feature = "debug_refcell")]
+                {
+                    if b.borrow.get() == 1 {
+                        self.first_caller.set(Some(crate::panic::Location::caller()));
```

Comment:
I was gonna say the following, but the closure would screw the `caller` call, right?

---

No idea if this is more "efficient", but an alternative here would be

`self.first_caller.update(|caller| caller.or(Some(crate::panic::Location::caller())))`

Reply:
Your suggestion won't work - `crate::panic::Location::caller()` will hit the boundary of the `|caller|` closure.